File: src/main/php/Easy/MajorityFrequencyGroup.php
<?php

declare(strict_types=1);

namespace Src\Easy;

class MajorityFrequencyGroup
{
    public function solve(string $s): string
    {
        $freq = [];
        $entries = [];
        foreach (str_split($s) as $char) {
            $freq[$char] = ($freq[$char] ?? 0) + 1;
        }
        foreach ($freq as $key => $value) {
            $entries[$value] = ($entries[$value] ?? '') . $key;
        }
        uksort($entries, function ($a, $b) use ($entries) {
            $c = strlen($entries[$a]);
            $d = strlen($entries[$b]);

            return ($c != $d) ? $d - $c : $b - $a;
        });

        return reset($entries);
    }
}

File: easy/majority_frequency_group.go
package easy

import (
	"maps"
	"slices"
)

func MajorityFrequencyGroup(s string) string {
	freq, entries := make(map[rune]int), make(map[int]string)
	for _, char := range s {
		freq[char]++
	}
	for k, v := range freq {
		entries[v] += string(k)
	}
	keys := slices.SortedFunc(maps.Keys(entries), func(a, b int) int {
		c, d := len(entries[a]), len(entries[b])
		if c != d {
			return d - c
		}
		return b - a
	})

	return entries[keys[0]]
}
